### PR TITLE
fix: issue #6866

### LINKED ISSUE #6866: Typing "use x;y" in a function crashes the extension
If you type `use x;y` in a function the VSCode extension will crash and you will need to restart the editor to get it working again.

Version info:
* OS: 64-bit Linux (OpenSUSE Tumbleweed)
* Flix 0.42.0
* vscode-flix v1.11.0

To reproduce:
1. Open a fresh folder in VSCode
2. Create a new main.flix file
3. Type the following code in it:
```
def main(): Unit =
    use x;y
```
4. As soon as you type the `y`, the extension crashes with error message like this:

```
❌ -- Parse Error -------------------------------------------------- file:///tmp/tmp.bGqyyytxDw/main.flix

>> Parse Error: Unexpected end of input, expected UnaryOp1, UnaryOp2 or Ref (line 2, column 11):
    use x;
          ^

Syntactic Context: OtherExpr.

Exception in thread "ForkJoinPool-7-worker-24" java.util.NoSuchElementException: head of empty list
	at scala.collection.immutable.Nil$.head(List.scala:662)
	at scala.collection.immutable.Nil$.head(List.scala:661)
	at ca.uwaterloo.flix.language.phase.Resolver$.tryLookupQualifiedName(Resolver.scala:2751)
	at ca.uwaterloo.flix.language.phase.Resolver$.ca$uwaterloo$flix$language$phase$Resolver$$lookupQualifiedName(Resolver.scala:2797)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.visitExp$1(Resolver.scala:1020)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.resolve(Resolver.scala:1685)
	at ca.uwaterloo.flix.language.phase.Resolver$.$anonfun$resolveDef$1(Resolver.scala:518)
	at ca.uwaterloo.flix.util.Validation$.flatMapN(Validation.scala:502)
	at ca.uwaterloo.flix.language.phase.Resolver$.resolveDef(Resolver.scala:515)
	at ca.uwaterloo.flix.language.phase.Resolver$.visitDecl(Resolver.scala:365)
	at ca.uwaterloo.flix.language.phase.Resolver$.$anonfun$visitUnit$3(Resolver.scala:335)
	at ca.uwaterloo.flix.util.Validation$.$anonfun$fastTraverse$1(Validation.scala:263)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at ca.uwaterloo.flix.util.Validation$.fastTraverse(Validation.scala:262)
	at ca.uwaterloo.flix.util.Validation$.traverse(Validation.scala:216)
	at ca.uwaterloo.flix.language.phase.Resolver$.$anonfun$visitUnit$2(Resolver.scala:335)
	at ca.uwaterloo.flix.util.Validation$.flatMapN(Validation.scala:502)
	at ca.uwaterloo.flix.language.phase.Resolver$.visitUnit(Resolver.scala:332)
	at ca.uwaterloo.flix.language.phase.Resolver$.$anonfun$run$9(Resolver.scala:111)
	at ca.uwaterloo.flix.util.ParOps$.$anonfun$parMap$2(ParOps.scala:49)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

### PR DESCRIPTION
**Pull request by magnus-madsen**
Fixes #6866

This also improves exception handling on `ParOps`

### CODE REVIEW

**Comment by magnus-madsen** [comment]
!!!jar

File: main/src/ca/uwaterloo/flix/util/ParOps.scala [hunk 1]
```diff
@@ -41,19 +41,32 @@ object ParOps {
...
     var idx = 0
     for (elm <- xs) {
       val i = idx // Ensure proper scope of i.
       flix.threadPool.execute(() => {
-        out(i) = f(elm)
-        latch.countDown()
+        try {
+          out(i) = f(elm)
+        } catch {
+          case ex: Throwable => thrown = ex
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/util/ParOps.scala:56]: Should we be catching all Throwables here or only exceptions?

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 2]
```diff
@@ -2740,6 +2740,11 @@ object Resolver {
     * Looks up the qualified name in the given root.
     */
   private def tryLookupQualifiedName(qname0: Name.QName, env: ListMap[String, Resolution], ns0: Name.NName, root: NamedAst.Root): Option[List[NamedAst.Declaration]] = {
+    // Fail if the name is un-qualified.
+    if (qname0.isUnqualified) {
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:2744]: This seems like a logical error rather than a runtime error.

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:2744]: Looking at the issue: `use x` should have a specific error like "can only use qualified names". But returning None here gives a NotFound error

File: main/src/ca/uwaterloo/flix/util/ParOps.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/util/ParOps.scala:56]: All throwables.

**Pull request closed without merging**